apps/wallet/src/utils/transaction/common.ts

```diff
@@ -42,9 +43,17 @@ export const checkValidAddress = (keyStr: string, network: Networks) => {
 			const { AccountAddress } = TxnBuilderTypes;
 			return { valid: AccountAddress.isValid(keyStr), message: '' };
 		}
-		return { valid: false, message: 'Unsupported network ' + network };
+		return {
+			valid: true,
+			message: '',
+		};
 	} catch (error) {
-		return { valid: false, message: (error as Error).message };
+		return {
+			valid: false,
+			message: network
+				? `Wrong [${capitalize(network)}] wallet address. Please check again.`
+				: (error as Error).message,
+		};
```

Comment:
to make things clear, I think we should refactor it to remove the valid flag, just return a string if It has error (error message)

Reply:
there is a case where `message` is `''` but it is not valid (as we update the checking condition right in the input phase), e.g. empty amount or empty recipient

Reply:
what next if `message === ''` but it is not valid

Reply:
It might confuse because of using two fields for a single purpose as error checking. If the message is not empty, it still is invalid (despite the valid having true value)

Reply:
You should have an alter convention/pattern to make it clean